src/Balances.jsx

```diff
@@ -1,12 +1,12 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useState, useMemo } from "react";
 import { Table, Grid } from "semantic-ui-react";
 
 export default function Balances(props) {
   const { api, keyring } = props;
   const accounts = keyring.getPairs();
-  const addresses = accounts.map(account => account.address);
   const accountNames = accounts.map(account => account.meta.name);
   const [balances, setBalances] = useState({});
+  const addresses = useMemo(() => accounts.map(account => account.address),[accounts]);
```

Comment:
Doesn't work. Just do a `console.log` in the `useEffect` and you'll see. `useMemo` is memoizing every single time, because it thinks `accounts` changes. Passing an empty array here is a hack because it will freeze the addresses and not re-render when `accounts` changes... so still not sure about what's best.